### PR TITLE
Work with broom 0.7.0

### PR DESCRIPTION
**Pull request by hideaki**
# Description
Making it work with broom 0.7.0.

# Checklist
Make sure you have performed following items before submitting this pull request.
If not, please describe the reason.  

- [ ] Add test cases for this fix/enhancement
- [ ] Pass devtools::check()
- [ ] Pass devtools::test()
- [ ] Test installing from github
- [x] Tested with Exploratory

### CODE REVIEW

File: R/broom_wrapper.R [hunk 1]
```diff
@@ -435,6 +513,21 @@ prediction <- function(df, data = "training", data_frame = NULL, conf_int = 0.95
...
         paste0("broom::augment(m, data = df, ", aug_args, ")")
       }
+      # From broom 0.7.0 augment.glm, if data argument is different from the one inside the model due to, for example, NA rows, error like following is raised.
+      #
+      # Assigned data `predict(x, newdata, type = type.predict) %>% unname()` must be compatible with existing data.
+      #
+      # To avoid it, do not specify data argument. Let's do this only for glm since, for example, ranger needs it to get actual prediction on training data instead of prediction on OOB data.
+      # Another option for workaround was to use newdata argument. But result from it does not have metrics like residuals, standardised_residuals, hat, residual_standard_deviation, or cooks_distance.
+      # Drawback of no argument is, the result misses columns in the original data that are not used for the model.
+      # TODO: Check if this applies to other models. What about lm??
```
Comment (kei51e) [inline R/broom_wrapper.R:523]: Please check this TODO:

File: tests/testthat/test_broom_wrapper.R [hunk 2]
```diff
@@ -350,51 +352,53 @@ test_that("test prediction binary", {
...
-  test_data[["rand"]] <- runif(nrow(test_data), min=-5, max = 5)
-
-  model_data <- build_model(test_data, model_func = mgcv::gam, formula = rand ~ s(DISTANCE), test_rate = 0.2)
-
-  prediction_ret <- prediction(model_data)
-
-  expect_true(any(colnames(prediction_ret) %in% "conf_low"))
-  expect_true(any(colnames(prediction_ret) %in% "conf_high"))
-})
+# This case is not working since broom 0.7.0, but it seems we don't use loess this way either. Remove it at some point.
```
Comment (kei51e) [inline tests/testthat/test_broom_wrapper.R:355]: Please remove.

Comment (hideaki) [reply tests/testthat/test_broom_wrapper.R:355]: Removed it.

File: R/broom_wrapper.R (hunk 1 above)
Comment (hideaki) [reply R/broom_wrapper.R]: lm had the same issue. Added the same handling as glm case, and added test for it too.

**Pull request merged**